### ISSUE TITLE
Chest inventory has too few player inventory slots

### ISSUE THREAD

**Issue by Monniasza** [minetest/minetest_game#3031]
When opening a chest, the chest doesn't show the last 4 slots of the player's inventory. It is annoying to not see items that are there.

Chest inventory (it omits 4 last player inventory slots):
![screenshot_20230506_104204](https://user-images.githubusercontent.com/37037712/236613869-cb5de988-9251-423c-9901-4fce2a31cfc2.png)
Player inventory (it has items invisible in the chest formspec):
![screenshot_20230506_104208](https://user-images.githubusercontent.com/37037712/236613872-f09d98be-a465-4088-9d30-d370eaf10e8c.png)

**Comment by Monniasza** [comment]
Additionally, the last hotbare slot is shown as if it were a regular slot.

**Comment by sfan5** [comment]
Wouldn't it be the responsibility of the mod that increased hotbar size to also make the necessary changes so that chests display correctly?

**Comment by Monniasza** [comment]
I've not changed the hotbar size, it is default size.

**Comment by sfan5** [comment]
No, the default hotbar size is 8 and it shows identical in the inventory as in a chest. I've just confirmed this myself.

**Comment by appgurueu** [comment]
It isn't just the hotbar size; it looks like a mod is extending your inventory to be 9*4 rather than 8*4. It would be the responsibility of this mod to deal with chest formspecs, although we should maybe provide an API to make this easier.

**Comment by rollerozxa** [comment]
i3 extends the inventory width from the default 8 to 9.

**Comment by Monniasza** [comment]
The game should expose an API to adjust and get player inventory capacity and hotbar size, which should be used by formspecs and could be changed by mods.

**Comment by appgurueu** [comment]
The engine already exposes such APIs. The hotbar (which also is engine configurable) is IMO unrelated; hotbar size *could* be used as inventory display width, but it shouldn't be.

I think it should be sufficient to just make all formspecs showing the player inventory variable (and parameterize them in terms of the player inventory size). This won't be particularly pretty (esp. for 9*4 slots, since the last row will have just 4 slots then), but it will be better than just hiding slots.

**Comment by Wuzzy2** [comment]
This is clearly a problem with the mods destroying the game too much. Please complain to the respective mod authors to fix these bugs.

I recommend to close this issue as invalid.

**Comment by appgurueu** [comment]
> This is clearly a problem with the mods destroying the game too much. Please complain to the respective mod authors to fix these bugs.

Indeed it is; however the issue on our end is that we don't really provide APIs to allow mod authors to change the inventory size without having to significantly duplicate MTG code. Strictly speaking such "dehardcoding" could be considered a feature though.
